src/service/performance-impl.js

```diff
@@ -189,6 +201,9 @@ export class Performance {
           'first-input'
         );
       }
+    } else {
+      // If no Performance Observer, resolve fcp promise
+      this.fcpResolve_();
```

Comment:
nit: let's try to avoid undefined, resolve with `null` instead.

Reply:
Is 'first-contentful-paint' always supported by performance observer. I realized there is a `supportedEntryTypes`. Could you please confirm. Thanks

Reply:
It seems that 'first-contentful-paint' is not always supported by performance observer. When I did integration tests in Safari, the performance observer was supported, but 'first-contentful-paint' was not recorded.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry/entryType

Is it `first-paint` or `first-contentful-paint`?

Reply:
This won't be needed with the platform solution